### ISSUE TITLE
Support for 64-bit data format ("CDF5") in scipy.io.netcdf_file

### ISSUE THREAD

**Issue by shoyer** [scipy/scipy#9162]
The 64-bit data format (also called "CDF5") is a variation of netCDF3 that adds support for additional data types (int64 and unsigned integers) and records with more than 2 billion elements:
http://cucis.ece.northwestern.edu/projects/PnetCDF/CDF-5.html

The format is supported by the netCDF-C library and netCDF4-python, but not yet officially documented in the netCDF spec -- apparently this is an oversight:
https://www.unidata.ucar.edu/mailing_lists/archives/netcdfgroup/2018/msg00067.html

It would be nice to support this in scipy for writing bigger files with a full complement of data types.

**Comment by nbren12** [comment]
@dschmitz89  Thanks for pointing out this issue. 

> Does https://github.com/Unidata/netcdf4-python maybe suit your needs? In general, this seems much better maintained than scipy.io.netcdf.

It's quite a small change to add support for the CDF-5 format (maybe 10-20 lines diff) to the scipy codebase. scipy.io.netcdf is a nice alternative to the netCDF4 libray for a few reasons. For example, netCDF4 doesn't support reading from file-like objects, multithreading is flaky, and it has numerous heavy dependencies. So a a simple pure python parser for the simpler versions of the netCDF format is quite handy.

**Comment by shoyer** [comment]
I agree, this is a minimal addition that would be worth landing in scipy. I would be happy to review this code.

Note that `scipy.io.netcdf_file` should also be updated to clarify that it does not write netCDF 4 files.
